### PR TITLE
[USM] Fix loading issue on some kernel version

### PR DESCRIPTION
**Pull request by Yumasi**
### What does this PR do?

This PR fixes a loading issue due to a kernel bug with packed enums, that are used in our HTTP2 monitoring. The bug was introduced in kernel 4.18 and fixed in 4.19.114, 5.4.29, 5.5.14 and 5.6.

### Describe how to test/QA your changes

The bug can be reproduced by running grpc tests on a 5.4 kernel, without the fix:
`inv -e system-probe.test --skip-linters -p ./pkg/network/protocols/grpc`
One should see errors such as:
```
Error: could not instantiate http monitor: error initializing http ebpf program: verifier error loading eBPF programs: map http2_static_table: load BTF: load btf: invalid argument: [8] STRUCT (anon) size=2 vlen=2: key type_id=6 bits_offset=0 Member exceeds struct_size (34 line(s) omitted)
```
After the fix, the tests should pass.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/ebpf/c/protocols/http2/decoding-defs.h [hunk 1]
```diff
@@ -84,7 +84,7 @@ typedef struct {
     __u64 request_started;
 
     __u16 response_status_code;
-    __u8 request_method;
+    __u32 request_method;
```
Comment (guyarb) [inline pkg/network/ebpf/c/protocols/http2/decoding-defs.h:87]: why is it required?

**Review by usamasaqib — APPROVED** [review]

File: pkg/network/ebpf/c/protocols/http2/decoding-defs.h (hunk 1 above)
Comment (Yumasi) [reply pkg/network/ebpf/c/protocols/http2/decoding-defs.h:87]: The request value is compared with static table values, which is now an u32 with the removal of the packed attribute. You can't compare u8 and u32 in Golang.

Comment (guyarb) [reply pkg/network/ebpf/c/protocols/http2/decoding-defs.h:87]: please open a jira ticket on that issue, we should fix it in a separate PR

**Review by guyarb — APPROVED** [review]

File: pkg/network/ebpf/c/protocols/http2/decoding-defs.h (hunk 1 above)
Comment (Yumasi) [reply pkg/network/ebpf/c/protocols/http2/decoding-defs.h:87]: Done. The ticket is USMO-413.

**Pull request merged**